packages/calcite-components/src/components/list/list.tsx

```diff
@@ -532,6 +532,7 @@ export class List extends LitElement implements InteractiveComponent, SortableCo
       return;
     }
 
+    event.stopPropagation();
```

Comment:
This and the block below (and any other common logic) would also be good to include in https://github.com/Esri/calcite-design-system/issues/11849.